tool/gravity/cli/update.go

```diff
@@ -181,22 +183,51 @@ will be returned to the "active" state.
 	return nil
 }
 
-func checkCanUpdate(site ops.Site) error {
-	gravityPackage, err := site.App.Manifest.Dependencies.ByName(constants.GravityPackage)
+func checkCanUpdate(cluster ops.Site, operator ops.Operator, manifest schema.Manifest) error {
+	existingGravityPackage, err := cluster.App.Manifest.Dependencies.ByName(constants.GravityPackage)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	supportsUpdate, err := supportsUpdate(*gravityPackage)
+	supportsUpdate, err := supportsUpdate(*existingGravityPackage)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 	if !supportsUpdate {
 		return trace.BadParameter(`
 Installed runtime version (%q) is too old and cannot be updated by this package.
 Please update this installation to a minimum required runtime version (%q) before using this update.`,
-			gravityPackage.Version, defaults.BaseUpdateVersion)
+			existingGravityPackage.Version, defaults.BaseUpdateVersion)
+	}
+
+	docker := manifest.SystemOptions.DockerConfig()
+	if docker == nil {
+		// No changes
+		return nil
 	}
+
+	existingDocker := cluster.ClusterState.Docker
+	if existingDocker.IsEmpty() {
```

Comment:
Should not similar check also be added on the server side? In `checkUpdateParameters`. That's where the check that didn't allow switching drivers used to be.